cmd/edit/externalauthproviders/cmd.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalauthproviders
+
+import (
+	"os"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	name                             string
+	issuerName                       string
+	issuerAudiences                  []string
+	issuerUrl                        string
+	issuerCaFile                     string
+	claimMappingGroupsClaim          string
+	claimMappingUsernameClaim        string
+	claimValidationRuleClaim         string
+	claimValidationRuleRequiredValue string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "external-auth-providers",
+	Aliases: []string{"externalauthproviders", "externalauthprovider", "external-auth-provider"},
+	Short:   "Edit external authentication providers",
+	Long:    "Edit external authentication providers on a cluster.",
+	Example: `  # Set issuer name 'issuer1' on cluster 'mycluster'
+  rosa edit external-auth-providers --issuer-name=issuer1 --cluster=mycluster`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddClusterFlag(Cmd)
+
+	flags.StringVar(
+		&args.name,
+		"name",
+		"",
+		"Name for the set of external authentication providers.",
+	)
+
+	flags.StringVar(
+		&args.issuerName,
+		"issuer-name",
+		"",
+		"Issuer name.",
+	)
+
+	flags.StringArrayVar(
+		&args.issuerAudiences,
+		"issuer-audiences",
+		nil,
+		"Names of audiences for the issuer.",
+	)
+
+	flags.StringVar(
+		&args.issuerUrl,
+		"issuer-url",
+		"",
+		"The url link for the issuer.",
+	)
+
+	// Is this a path?
+	flags.StringVar(
+		&args.issuerCaFile,
+		"issuer-ca-file",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimMappingGroupsClaim,
+		"claim-mapping-groups-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimMappingUsernameClaim,
+		"claim-mapping-username-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimValidationRuleClaim,
+		"claim-validation-rule-claim",
+		"",
+		"TBD.",
+	)
+
+	flags.StringVar(
+		&args.claimValidationRuleRequiredValue,
+		"claim-validation-rule-required-value",
+		"",
+		"TBD.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	if cluster.Hypershift().Enabled() {
```

Comment:
NP : Invert this check, it will remove the need for the `else` and we can return early if the cluster is not HCP. 

Potentially might be worth moving this logic to a helper function for easier testing and improved clarity.